src/org/traccar/protocol/WatchProtocolDecoder.java

```diff
@@ -296,6 +308,14 @@ protected Object decode(
 
         } else if (type.equals("TK")) {
 
+            if (buf.readableBytes() == 1) {
+                byte result = buf.readByte();
+                if (result != '1') {
+                    LOGGER.error(type + "," + result);
```

Comment:
This is reserved for some serious errors, not protocols decoding issues. Also, what is the purpose of this check?

Reply:
The 1st if is needed because otherwise we are saving 1 byte long amr files when the device responds to a voice message we sent.